crates/bevy_asset/src/server/info.rs

```diff
@@ -506,7 +514,7 @@ impl AssetInfos {
             if let Some(info) = self.get_mut(id) {
                 info.loading_dependencies.remove(&loaded_asset_id);
                 if info.loading_dependencies.is_empty() {
-                    // send dependencies loaded event
+                    // send dependencies loaded event (wouldn't this overwrite previous Failed state?)
```

Comment:
This seems broken as is. Example: A depends on B and C.

1. B fails to load. 
2. Line 624: B removed from `A.loading_dependencies` and 
3. Line 628: `A.dep_load_state` is set to Failed
4. C successfully loads
5. Line 515: C removed from `A.loading_dependencies` (now empty)
6. Line 518:  `A.dep_load_state` overwritten to Loaded

Reply:
Added a new unit test that confirmed the previous impl was overwriting previously failed dependencies. Added this check to fix it.